src/System.Private.ServiceModel/src/System/ServiceModel/Channels/WebSocketTransportDuplexSessionChannel.cs

```diff
@@ -536,27 +536,9 @@ private void AsyncReceiveCancelled()
                 return message;
             }
 
-            // TODO: As we're waiting blocking on a task anyway, should just call ReceiveAsync and block on that task.
             public Message Receive(TimeSpan timeout)
             {
-                bool waitingResult = _receiveTask.Task.WaitWithTimeSpan(timeout);
-                ThrowOnPendingException(ref _pendingException);
-
-                if (!waitingResult)
-                {
-                    throw FxTrace.Exception.AsError(new TimeoutException(
-                               SR.Format(SR.WaitForMessageTimedOut, timeout),
-                               TimeoutHelper.CreateEnterTimedOutException(timeout)));
-                }
-
-                Message message = GetPendingMessage();
-
-                if (message != null)
-                {
-                    StartNextReceiveAsync();
-                }
-
-                return message;
+                return ReceiveAsync(timeout).WaitForCompletionNoSpin();
```

Comment:
Question only: are we able to address the TODO now because of the change, or is this just something you noticed we needed to clean-up?  Just curious.

And why does it call WaitForCompletionNoSpin() without the timeout rather than the one that takes a timeout?  Is that because the ReceiveAsync's timeout will guarantee this flavor of WaitForCompletionNoSpin will complete if a timeout occurs?